packages/tasks/src/tasks/text-generation/spec/output.json

```
@@ -10,43 +10,42 @@
 			"description": "The generated text"
 		},
 		"details": {
-			"description": "When enabled, details about the generation",
-			"title": "TextGenerationOutputDetails",
-			"allOf": [
-				{ "$ref": "#/$defs/SequenceDetails" },
-				{
-					"type": "object",
-					"properties": {
-						"best_of_sequences": {
-							"type": "array",
-							"description": "Details about additional sequences when best_of is provided",
-							"items": {
-								"allOf": [
-									{ "$ref": "#/$defs/SequenceDetails" },
-									{
-										"type": "object",
-										"properties": {
-											"generated_text": {
-												"type": "integer",
-												"description": "The generated text"
-											}
-										},
-										"required": ["generated_text"]
-									}
-								]
-							}
-						}
-					}
-				}
-			]
+			"$ref": "#/$defs/Details",
+			"description": "When enabled, details about the generation"
 		}
 	},
 	"required": ["generated_text"],
-
 	"$defs": {
+		"FinishReason": {
+			"type": "string",
+			"title": "TextGenerationFinishReason",
+			"description": "The reason why the generation was stopped.",
+			"oneOf": [
+				{ "const": "length", "description": "The generated sequence reached the maximum allowed length" },
+				{ "const": "eos_token", "description": "The model generated an end-of-sentence (EOS) token" },
+				{ "const": "stop_sequence", "description": "One of the sequence in stop_sequences was generated" }
```

nit: maybe let's add the value in the description so the generated docstring is more helpful
```suggestion
				{ "const": "length", "description": "length: The generated sequence reached the maximum allowed length" },
				{ "const": "eos_token", "description": "eos_token: The model generated an end-of-sentence (EOS) token" },
				{ "const": "stop_sequence", "description": "stop_sequence: One of the sequence in stop_sequences was generated" }
```

---

Definitely a good idea!